### PR TITLE
feat: use fedimint-bin-tests in rust tests

### PR DESCRIPTION
**Pull request by Maan2003**

### CODE REVIEW

**Review by dpc — APPROVED** [review]

**Review by justinmoon — DISMISSED** [review]

File: fedimint-bin-tests/src/main.rs [hunk 1]
```diff
@@ -1109,6 +1124,15 @@ struct Args {
     command: Cmd,
 }
 
+async fn write_ready_file<T>(result: Result<T>) -> Result<T> {
+    let ready_file = env::var("FM_READY_FILE")?;
+    match result {
+        Ok(_) => fs::write(ready_file, "READY").await?,
```
Comment (dpc) [inline fedimint-bin-tests/src/main.rs:1130]: I just banned these via semgrep like yesterday or day before. :D

Comment (Maan2003) [reply fedimint-bin-tests/src/main.rs:1201]: does CI check precommit hooks? 
I don't run precommit hooks locally because my emacs just lags _a lot_.

Comment (Maan2003) [reply fedimint-bin-tests/src/main.rs:1201]: looks like you banned std::fs::write, not tokio::fs::write

File: fedimint-bin-tests/src/main.rs [hunk 2]
```diff
@@ -1047,6 +1047,35 @@ struct DevFed {
     esplora: Esplora,
 }
 
+#[allow(unused)]
+struct BaseDaemons {
```
Comment (justinmoon) [inline fedimint-bin-tests/src/main.rs:1051]: Maybe a better name might be `ExternalDaemons` or something. `fedimintd` is really the "base" one in our system, but grouping these out in principle sounds great 👍

**Review by justinmoon — APPROVED** [review]
Can fix the nits tomorrow.

File: fedimint-bin-tests/src/main.rs [hunk 3]
```diff
@@ -1047,6 +1047,35 @@ struct DevFed {
...
+#[allow(unused)]
+struct ExternalDaemons {
+    bitcoind: Bitcoind,
+    cln: Lightningd,
+    lnd: Lnd,
+    electrs: Electrs,
+    esplora: Esplora,
+}
+
+async fn base_daemons(process_mgr: &ProcessManager) -> Result<ExternalDaemons> {
```
Comment (Maan2003) [inline fedimint-bin-tests/src/main.rs:1059]: missed rename

**Review by dpc — DISMISSED** [review]

**Comment by Maan2003** [comment]
this failed CI in past. but now has succeed for 4 time consecutively now and there is nothing in this that PR should break tests.
So I am marking this as ready to review.

**Review by justinmoon — APPROVED** [review]
Beautiful. It's happening!

**Pull request merged**